### PR TITLE
Fix employees can see postings outside of branch

### LINKED ISSUE #647: Employee can see postings outside of their branches in their homepage
employees should not be able to see postings outside of their branches anywhere

### PR DESCRIPTION
**Pull request by briantu**
## Ticket link

Closes #647 

## Implementation description
* Filter by user branches in `postings` resolver if user is volunteer or employee
  * If employee also only return scheduled or past postings
  * If volunteer also only return unscheduled or scheduled postings
  * Removed old way of doing this for volunteers where we pass an optional userId parameter
* Updated `VolunteerPostingsAvailabilities` page to redirect to not found page on posting error
* Updated `SchedulePostingsPage` to redirect to not found page on posting error
* Updated `branches` query only returns branches that user is in

## Steps to test
1. Log in as employee
2. Check that you can only see postings of branches you are in
3. Log in as volunteer, and check that you can also only see postings of branches you are in

## What should reviewers focus on?
* Code correctness

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by MatoPlus — COMMENTED** [review]
Awesome work! I'll test this and update with a review in a bit!

File: backend/utilities/dateUtils.ts [hunk 1]
```diff
@@ -44,3 +44,9 @@ export const getTodayForTZIgnoredUTC = (date: Date = new Date()): Date => {
     .startOf("day")
     .toDate();
 };
+
+export const isPast = (date: Date): boolean => {
+  return moment(moment(date).utc().format("YYYY-MM-DD")).isBefore(
```
Comment (MatoPlus) [inline backend/utilities/dateUtils.ts:49]:
```suggestion
  return moment(moment(date).format("YYYY-MM-DD")).isBefore(
```

I don't think UTC is needed

File: backend/services/implementations/postingService.ts [hunk 2]
```diff
@@ -238,18 +212,30 @@ class PostingService implements IPostingService {
...
-          volunteer.branches.filter(
+          volunteerOrEmployee.branches.filter(
             (branch: BranchResponseDTO) =>
               Number(branch.id) === posting.branch.id,
           ).length === 0
         ) {
           throw new Error(`User is not part of ${posting.branch.name} branch.`);
         }
+        if (
+          user.role === Role.EMPLOYEE &&
```
Comment (MatoPlus) [inline backend/services/implementations/postingService.ts:229]: do we have a filter for volunteers to return err for draft postings?

Comment (briantu) [reply backend/services/implementations/postingService.ts:229]: Oh, I didn't do any filter for volunteers. Which posting statuses should volunteers only be able to see?

File: backend/services/implementations/postingService.ts [hunk 3]
```diff
@@ -238,18 +212,30 @@ class PostingService implements IPostingService {
...
+          !(
+            isPast(posting.endDate) ||
+            isPostingScheduledBySignups(
+              posting.shifts.flatMap((shift) => shift.signups),
+            )
```
Comment (MatoPlus) [inline backend/services/implementations/postingService.ts:234]:
```suggestion
          (posting.status === "DRAFT" ||
            !isPostingScheduledBySignups(
              posting.shifts.flatMap((shift) => shift.signups),
            )
```

File: backend/services/implementations/postingService.ts (hunk 2 above)
Comment (MatoPlus) [reply backend/services/implementations/postingService.ts:229]: as discussed, volunteer should not be able to see past or draft.

File: backend/utilities/dateUtils.ts (hunk 1 above)
Comment (MatoPlus) [reply backend/utilities/dateUtils.ts:49]: update: we don't need this function. Let's remove it.

File: backend/services/implementations/postingService.ts (hunk 2 above)
Comment (briantu) [reply backend/services/implementations/postingService.ts:229]: volunteers can only see unscheduled and scheduled postings

File: backend/utilities/dateUtils.ts [hunk 4]
```diff
@@ -44,3 +44,9 @@ export const getTodayForTZIgnoredUTC = (date: Date = new Date()): Date => {
...
+    moment(moment().format("YYYY-MM-DD")),
+  );
+};
```
Comment (MatoPlus) [inline backend/utilities/dateUtils.ts:52]:
```suggestion
export const isPast = (date: Date): boolean => {
  return moment(getTodayForTZIgnoredUTC(date)).isBefore(
    moment(getTodayForTZIgnoredUTC()),
  );
};
```

**Review by MatoPlus — APPROVED** [review]
LGTM! Thanks for fixing all these visibliity issues!

**Pull request merged**